projecttacoma/fqm-testify
# Population comparison table

# Summary
This PR adds a table to the `MeasureHighlightingPanel` that displays a comparison of the desired populations and actual populations that the patient falls under. 

## New Behavior
When a patient is selected, a population comparison table is rendered in the `MeasureHighlightingPanel` above the highlighted logic. This table has two columns of values: Desired and Actual. The rows are the populations that exist on the uploaded measure bundle. The values in the Desired row represent whether or not that population is included in the Desired population (1 for included, 0 for not). The values in the Actual row represent whether or not that population exists on the measure calculation measure report (1 for exists, 0 for does not). Additionally, when rows of the table match, they are highlighted green. When the rows do not match, they are highlighted red.

## Code Changes
- `components/utils/PopulationMultiSelect.tsx` - moved the `getMeasurePopulations` function to a new file (`MeasurePopulations.tsx`) so it can get reused.
- `components/calculation/MeasurePopulations.tsx` - added the `getMeasurePopulations` function.
- `components/calculation/MeasureHighlightingPanel.tsx` - added functions in order to populate the population comparison table, renders the population comparison table.
- `__tests__/components/calculation/MeasureHighlightingPanel.test.tsx` - added unit test for the population comparison table.

# Testing Guidance
1. Run `npm run check` to ensure that all unit tests pass and there are no lint/prettier errors or warnings.
2. Upload a measure bundle and create or upload a patient (make sure to do both at some point).
3. Select the patient without any desired populations added and make sure that the desired population column is populated with all 0s.
4. Add desired populations and ensure that the correct values in the table are changing. 
5. Add measure resources to get the patient in any of the populations and ensure that those values change in the table as well.
7. I am purposely leaving this kind of open ended so that any bugs that I am not thinking of can be caught, but let me know if you have any questions or need help!
